src/dune_rules/obj_dir.mli

```diff
@@ -50,8 +50,11 @@ val melange_dir : 'path t -> 'path
 
 val all_cmis : 'path t -> 'path list
 
-(** The public compiled cmi file directory *)
-val public_cmi_dir : 'path t -> 'path
+(** The public compiled cmi file directory for ocaml *)
+val public_cmi_ocaml_dir : 'path t -> 'path
+
+(** The public compiled cmi file directory for melange *)
+val public_cmi_melange_dir : 'path t -> 'path
```

Comment:
Perhaps it would be simpler to keep one function and add an argument:

```
[`Melange | `OCaml]
```

Reply:
This function is used only once for Melange case, not sure about the benefits of adding a type just to be able to share with OCaml code paths, but we can add it in the future if we stumble upon more cases like this.

Reply:
That's why we have polymorphic variants, so that we don't need to introduce new types :)

In any case, it's not an important issue.